tests/Sharpliner.Model.Tests/GitHub/WorkflowTests.cs

```diff
@@ -0,0 +1,154 @@
+                    }
+                }
+            };
+
+            Assert.NotNull(w);
+            Assert.NotNull(w.On.Manuals.WorkflowDispatch);
+            Assert.Null(w.On.Manuals.RepositoryDispatch);
+            Assert.Empty(w.On.Schedules);
+            Assert.Empty(w.On.Webhooks);
+        }
+
+        [Fact]
+        public void Workflow_Manual_RepositoryDispatch()
+        {
+            var w = new Workflow
+            {
+                On =
+                {
+                   Manuals =
+                   {
+                       RepositoryDispatch = new () {
+                           Activities = { "Opened", "MyActivity"}
+                       }
+                   }
+                }
+            };
+
+            Assert.NotNull(w);
+            Assert.Null(w.On.Manuals.WorkflowDispatch);
+            Assert.NotNull(w.On.Manuals.RepositoryDispatch);
+            Assert.Empty(w.On.Schedules);
+            Assert.Empty(w.On.Webhooks);
+        }
+
+        [Fact]
+        public void Workflow_Single_Schedule()
+        {
+            var w = new Workflow
+            {
+                On =
+                {
+                    Schedules =
+                    {
+                        new ("'*/30 5,17 * * *'")
+                    }
+                }
+            };
+            Assert.NotNull(w);
+            Assert.Null(w.On.Manuals.WorkflowDispatch);
+            Assert.Null(w.On.Manuals.RepositoryDispatch);
+            Assert.Single(w.On.Schedules);
+            Assert.Empty(w.On.Webhooks);
+        }
+
+        [Fact]
+        public void Workflow_Several_Sechedule()
+        {
+            var w = new Workflow
+            {
+                On =
+                {
+                    Schedules =
+                    {
+                        new ("'*/30 5,17 * * *'"),
+                        new ("'*/30 1,17 * * *'")
+                    }
+                }
+            };
+            Assert.NotNull(w);
+            Assert.Null(w.On.Manuals.WorkflowDispatch);
+            Assert.Null(w.On.Manuals.RepositoryDispatch);
+            Assert.Equal(2, w.On.Schedules.Count);
+            Assert.Empty(w.On.Webhooks);
+        }
+
+        [Fact]
+        public void Workflow_Single_Webhook()
+        {
+            var w = new Workflow()
+            {
+                On =
+                {
+                    Webhooks =
+                    {
+                        new CheckRun
+                        {
+                            Activities =  { CheckRun.Activity.Completed, CheckRun.Activity.RequestedAction }
+                        }
+                    }
+                }
+            };
+            Assert.NotNull(w);
+            Assert.Null(w.On.Manuals.WorkflowDispatch);
+            Assert.Null(w.On.Manuals.RepositoryDispatch);
+            Assert.Empty(w.On.Schedules);
+            Assert.Single(w.On.Webhooks);
+        }
+
+        [Fact]
+        public void Workflow_Several_webhook()
```

Comment:
I am not sure unit tests like this one won't just add extra work to any refactoring which we might do in the early stages?

Reply:
I like them because they allow me to test the API... and I want to also test the static analysis using the tests :) I don't mind doing it.